datafusion/core/src/datasource/file_format/csv.rs

```diff
@@ -454,7 +455,9 @@ impl CsvFormat {
                             .has_header
                             .unwrap_or(state.config_options().catalog.has_header),
                 )
-                .with_delimiter(self.options.delimiter);
+                .with_delimiter(self.options.delimiter)
+                // Support literal NULL or empty string as null
+                .with_null_regex(Regex::new(r"^NULL$|^$").unwrap());
```

Comment:
Shouldn't this be a configuration option instead?
Most csv writers will use an empty string (`,,`) for null which also makes sense as default.
Having the regex enabled might also impact performance?

Reply:
A configuration option makes sense to me

I poked around for a csv querying benchmark and I can't find one. Perhaps we should write one 🤔

Reply:
So, to confirm:

I'll open up a separate PR to implement a benchmark.

When that gets merged I'll update this PR to use a configuration option and add the new config to the benchmark test to confirm if there is a regression.

Reply:
Agreed -- thank you @dhegberg

Reply:
I think it might make sense to make it an configuration regardless if there is a performance degradation or not. This is because the new behavior will be a breaking change and potentially undesired behavior for some users.